packages/flutter/lib/src/widgets/text_selection.dart

```diff
@@ -623,10 +627,15 @@ class TextSelectionOverlay {
     if (!renderObject.attached) {
       return;
     }
-    final Size handleSize = selectionControls!.getHandleSize(
-      renderObject.preferredLineHeight,
-    );
-    _dragStartPosition = details.globalPosition + Offset(0.0, -handleSize.height);
+
+    // [details.globalPosition] is the drag point, it needs to be shifted
+    // to the text position point that [RenderEditable] uses to get
+    // [TextPosition] and update selection for start handle correctly.
+    // Dx of text position point need to be consistent with dx of drag point.
+    // Dy of text position point need to be locate in the correct text line.
```

Comment:
See my comment above for simplifying this comment.

Reply:
Done.